daffodil-lib/src/main/scala/org/apache/daffodil/xml/XMLUtils.scala

```diff
@@ -748,16 +748,13 @@ object XMLUtils {
 Comparison failed.
 Expected (attributes stripped)
           %s
-Actual (attributes stripped)
+Actual (attributes ignored for diff)
```

Comment:
Do we have a test that verifies this ignore logic doesn't strip the xsi:nil attribute?  Just asking since Guichard had an issue related to that with EDIFACT recently, and I wanted to make sure the issue is not in the TDML comparison logic. I don't think it is, but wanted to check.

Reply:
I'm not sure if we have an explicit test for the XMLUtils xml comparison logic to make sure we can find xsi:nil differences. However, I have modified a nillable test to remove ``xsi:nil="true"`` from the expected infoset and got an error saying the expected infoset did not have the attribute, so I think our xsi:nil checking logic is correct. It's very possible that there's a edgecase or a bug somewhere, but the expected cases look correct to me. Attributes are ignored, but we special case and xsi:nil as expected.